Sources/Common/Utils/DependencyFactory.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  DependencyFactory.swift
+//
+//
+//  Created by Nityananda Zbil on 06.06.24.
+//
+
+import Foundation
+
+/// Decides the correct service for live- and test environments.
+///
+/// Example of a minimal implementation:
+///
+/// ```swift
+/// struct NumberFactory: DependencyFactory {
+///     static var liveValue: Int {
+///         1
+///     }
+/// }
+/// ```
+///
+/// Usage: `NumberFactory.shared`
+public protocol DependencyFactory {
+    associatedtype Value
+
+    static var shared: Value { get }
+
+    static var liveValue: Value { get }
+
+    static var testValue: Value { get }
+}
+
+public extension DependencyFactory {
+    static var shared: Value {
+        CommandLine.arguments.contains("-dependency-factory-test-value") ? Self.testValue : Self.liveValue
```

Comment:
https://github.com/ls1intum/artemis-ios/pull/112/commits/080bcddc9541c23aa6d56278ed37f7719523091d

Reply:
Hmm, `-Screenshots` more clearly stated the purpose of the argument since it could affect more parts of the app in the future and was also consistent with the implementation in apollon-ios. If something else needs to change in the future for Screenshots, we'd need another argument.
For consistency, I'd prefer to have them all the same. Maybe we keep `-Screenshots` and add your suggestion as a second option? That way we can expand what each option does in the future